### PR TITLE
Implements List View Functionality

### PR DESCRIPTION
**Pull request by AlexEntrepreneur**
### Summary of Changes

- Renamed `ListItems` Component to `List`
- Created a Reusable `List` Component
- Installed Axios
- Implemented `ExampleList` Component Using Axios
- Removed Unused `.gitnignore` File
- Added Tests For Each New Files

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -43,6 +44,7 @@
     "@testing-library/jest-dom": "^4.2.4",
     "@testing-library/react": "^10.2.1",
     "@testing-library/user-event": "^7.2.1",
-    "eslint-plugin-react-hooks": "^4.0.4"
+    "eslint-plugin-react-hooks": "^4.0.4",
```
Comment (ryan-hamblin) [inline package.json:47]: I love this idea, is this better served as a "DevDependencie" though?

File: src/index.js [hunk 2]
```diff
@@ -3,6 +3,7 @@ import ReactDOM from "react-dom";
 import { BrowserRouter, Route } from "react-router-dom";
 import { Security, LoginCallback } from "@okta/okta-react";
 
+import ExampleList from "./components/ExampleListComponent"
```
Comment (ryan-hamblin) [inline src/index.js:6]: The rest of these have semi-colons. Maybe we need to tweak the linter/prettier config to either do semi's or not. 

It also sparks the conversation of other opinionated conventions we ought to be using.

File: src/components/ExampleListComponent/RenderExampleList.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      ))
+    }
+  </div>
+)
+
+export default RenderExampleList;
+
+RenderExampleList.propTypes = {
+  data: PropTypes.arrayOf(
+    PropTypes.shape({
```
Comment (ryan-hamblin) [inline src/components/ExampleListComponent/RenderExampleList.js:23]: This is huge! Giving the students a lesson on `PropTypes` and `type checking`, in general, will go a LONG way IMO

File: src/api/index.js [hunk 4]
```diff
@@ -1,8 +1,14 @@
...
 
 const sleep = (time) =>
   new Promise((resolve) => {
     setTimeout(resolve, time);
   });
 
-export { sleep };
+const getExampleData = () => {
+  return axios.get(`https://jsonplaceholder.typicode.com/photos?albumId=1`)
+    .then(response => response.data)
```
Comment (ryan-hamblin) [inline src/api/index.js:11]: I'm going to be nit-picky here and ask for semi-colons until we standardize. I think they're good for learners to have to use. Again, our linter/prettier formatter config ought to pick these up. Setting up an issue on that now.

File: src/__tests__/RenderExampleList.test.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+      // We don't want to mutate the original data!
+      const newO = { ...o };
+      delete newO.id;
+      return newO;
+    });
+
+    // We can mock the console error method to check which errors the console returns 
+    // and to prevent errors from printing in our tests
+    // https://jestjs.io/docs/en/jest-object#jestspyonobject-methodname
+    const mockConsoleErrorMethod = jest.spyOn(console, 'error').mockImplementation();
```
Comment (ryan-hamblin) [inline src/__tests__/RenderExampleList.test.js:57]: LOVE <3 

This is the right example we need to give students on mocking API calls :)

File: src/__tests__/List.test.js [hunk 6]
```diff
@@ -0,0 +1,47 @@
+import React from "react";
+import { render, cleanup, act } from "@testing-library/react";
+
+import { List } from "../components/ReusableComponents";
+
+afterEach(() => {
+  cleanup();
+});
+
+const getItemsData = jest.fn(() => Promise.resolve([{id: 1}, {id: 2}, {id: 3}]))
```
Comment (ryan-hamblin) [inline src/__tests__/List.test.js:10]: Semi-colon :)

File: src/__tests__/List.test.js [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+const RenderItems = jest.fn((props) => props.data.map((item) => <li key={item.id}>Item</li>))
```
Comment (ryan-hamblin) [inline src/__tests__/List.test.js:11]: semi

File: src/components/ExampleListComponent/RenderExampleList.js [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+        <figure key={item.id}>
+          <img src={item.thumbnailUrl} alt={item.title} />
+          <figcaption>
+            <h3>{item.title}</h3>
+          </figcaption>
+        </figure>
+      ))
+    }
+  </div>
+)
```
Comment (ryan-hamblin) [inline src/components/ExampleListComponent/RenderExampleList.js:17]: semi

File: package.json (hunk 1 above)
Comment (AlexEntrepreneur) [reply package.json:47]: Github seemed to truncate the lines so it looked like this line was part of `dependencies` but this is currently a dev dependency.

**Pull request merged**